packages/frontend/src/contexts/Web3Context.tsx

```diff
@@ -270,9 +269,13 @@ const Web3ContextProvider: FC = ({ children }) => {
 
   // TODO: cleanup
   const checkConnectedNetworkId = useCallback(
-    async (networkId?: number): Promise<boolean> => {
+    async (networkId?: number | string): Promise<boolean> => {
```

Comment:
in what context or scenario will networkId be a `string`?

Reply:
i see in many places that we convert to number just prior to calling the method. this change adds some flexibility.

![Screen Shot 2022-03-07 at 9 54 17 AM](https://user-images.githubusercontent.com/18040654/157058455-bc8fd657-989b-4a87-85a7-c38b751b5263.png)

Reply:
the `Number` wrap here is legacy code. the config `networkId` should always be a number ([type](https://github.com/hop-protocol/hop/blob/94b2570eebe7c71da7ad9d8f657d5daf0aa8dc8d/packages/core/src/networks/types.ts#L3)) and `provider.getNetwork()` will always return `number` so i don't think there's need to have string type

Reply:
i see what's happening. actually, `sourceNetwork` in this particular usage is the `Network` model ([type](https://github.com/hop-protocol/hop/blob/develop/packages/frontend/src/models/Network.ts#L25)), which converts the original `networkId.toString()` ([here](https://github.com/hop-protocol/hop/blob/develop/packages/frontend/src/config/mainnet.ts#L27)). i just tested removing `.toString()` in that file, and it doesn't seem to break the app, as `Number` accepts `any`.

Reply:
nice, removing the string types is much cleaner